File: Reto 9 - Heterograma, Isograma y Pangrama/js/reto9.js
const ALPHABET = "abcdefghijklmnñopqrstuvwxyz";

function char_Counter(cadena) {
    let charCounter = {};

    for (let c of cadena) {
        if (charCounter[c]) {
            charCounter[c]++;
        } else {
            charCounter[c] = 1;
        }
    }
    return charCounter;
}

function isHeterogram(cadena) {
    let charCounter = char_Counter(cadena);

    for (let count in charCounter) {
        if (charCounter.hasOwnProperty(count) && charCounter[count] > 1) {
            return false;
        }
    }
    return true;
}

function isIsogram(cadena) {
    let charCounter = char_Counter(cadena);
    
    for (let count of Object.values(charCounter)) {
        if (count > 1) {
            return false;
        }
    }
    return true;
}

function isPangram(cadena) {
    let charCounter = char_Counter(cadena);

    if (Object.keys(charCounter).length === ALPHABET.length) {
        return true;
    } else {
        return false;
    }
}

function main() {
    let exampleText = "Benjamín pidió una bebida de kiwi y fresa. Noé, sin vergüenza, la más exquisita champaña del menú";

    console.log("Is Heterogram: " + isHeterogram(exampleText));
    console.log("Is Isogram: " + isIsogram(exampleText));
    console.log("Is Pangram: " + isPangram(exampleText));
}

main();

File: Reto 9 - Heterograma, Isograma y Pangrama/java/reto9.java
import java.util.HashMap;
import java.util.Map;

public class reto9 {
    
    private static final String ALPHABET = "abcdefghijklmnñopqrstuvwxyz";

    private static Map<Character, Integer> char_Counter(String cadena){
        Map<Character,Integer> charCounter = new HashMap<>();

        for(char c : cadena.toCharArray()){
            if(charCounter.containsKey(c)){
                charCounter.put(c,charCounter.get(c)+1);
            }
            else charCounter.put(c,1);
        }
        return charCounter;
    }

    public static boolean isHeterogram(String cadena){
        Map<Character,Integer> charCounter = char_Counter(cadena);

        for(int count: charCounter.values()){
            if(count>1) return false;
        }
        return true;
    }

    public static boolean isIsogram(String cadena){
        Map<Character,Integer> charCounter = char_Counter(cadena);
        for(int count: charCounter.values()){
            if(count > 1) return false;
        }
        return true;
    }

    public static boolean isPangram(String cadena){
        Map<Character,Integer> charCounter = char_Counter(cadena);

        if(charCounter.size() == ALPHABET.length()) return true;
        else return false;
    }
    
    public static void main(String[] args){
        String exampleText = "Benjamín pidió una bebida de kiwi y fresa. Noé, sin vergüenza, la más exquisita champaña del menú";

        System.out.println("Is Heterogram: " + isHeterogram(exampleText));
        System.out.println("Is Isogram: " + isIsogram(exampleText));
        System.out.println("Is Pangram: " + isPangram(exampleText));
    }
  
}
